jpa2ddl-core/src/main/java/com/devskiller/jpa2ddl/SchemaGenerator.java

```diff
@@ -116,43 +69,58 @@ void generate(GeneratorSettings settings) throws Exception {
-			SchemaProcessor schemaProcessor = schemaProcessorClass.getDeclaredConstructor().newInstance();
-			schemaProcessor.postProcess(connection, settings.getProcessorProperties());
+			databaseSchemaProcessor.processDatabase(settings, outputFile, jdbcDatabaseContainer.getJdbcUrl(), driverClassName, jdbcDatabaseContainer.getUsername(), jdbcDatabaseContainer.getPassword(), null);
+		} finally {
+			if (jdbcDatabaseContainer.isRunning()) {
+				jdbcDatabaseContainer.stop();
+			}
 		}
 	}
 
-	private Set<URL> getExistingUrls(Set<URL> urls) {
-		return urls.stream()
-				.filter(url -> {
-					try {
-						return new File(url.toURI()).exists();
-					} catch (URISyntaxException e) {
-						throw new RuntimeException(e);
-					}
-				})
-				.collect(Collectors.toSet());
+	private void handleEmbeddedDatabaseGeneration(GeneratorSettings settings, File outputFile) throws Exception {
+		EngineDecorator engineDecorator = EngineDecorator.getEngineDecorator(settings.getJpaProperties().getProperty(HIBERNATE_DIALECT));
+
+		String dbUrl = engineDecorator.decorateConnectionString(DB_URL);
+
+		databaseSchemaProcessor.processDatabase(settings, outputFile, dbUrl, "org.h2.Driver", "sa", "", engineDecorator::decorateDatabaseInitialization);
 	}
 
-	private String getDbUrl(EngineDecorator engineDecorator) {
-		return engineDecorator.decorateConnectionString(DB_URL);
+	private void handleMetadataGeneration(GeneratorSettings settings, File outputFile) throws Exception {
+		MetadataSources metadata = databaseSchemaProcessor.prepareMetadataSources(settings, outputFile, null, null, null);
+
+		SchemaExport export = new SchemaExport();
+		export.setFormat(settings.isFormatOutput());
+		export.setDelimiter(settings.getDelimiter());
+		export.setOutputFile(outputFile.getAbsolutePath());
+		export.execute(EnumSet.of(TargetType.SCRIPT), settings.getAction().toSchemaExportAction(), metadata.buildMetadata());
+	}
+
+	private Class<? extends JdbcDatabaseContainer> resolveJdbcContainerClass() {
+		Set<Class<? extends JdbcDatabaseContainer>> subTypesOf = classResolver.getSubTypesOf(JdbcDatabaseContainer.class);
+		if (subTypesOf.size() > 1) {
+			throw new IllegalStateException("More than one JdbcDatabaseContainer found");
+		} else if (subTypesOf.isEmpty()) {
+			throw new IllegalStateException("No JdbcDatabaseContainer found - please add dependency to proper test-containers module");
+		}
+
+		return subTypesOf.iterator().next();
 	}
 
 	private void validateSettings(GeneratorSettings settings) {
 		if (settings.getAction() == Action.UPDATE) {
 			if (settings.getOutputPath().exists() && !settings.getOutputPath().isDirectory()) {
 				throw new IllegalArgumentException("For UPDATE action outputPath must be a directory");
 			}
-			if (settings.getGenerationMode() != GenerationMode.DATABASE) {
+			if (settings.getGenerationMode() != GenerationMode.EMBEDDED_DATABASE && settings.getGenerationMode() != GenerationMode.CONTAINER_DATABASE) {
```

Comment:
Maybe some flag on `GenerationMode` to avoid `&&`? Also, the error message below is incorrect